gprofiler/kernel_messages.py

```diff
@@ -2,41 +2,32 @@
 # Copyright (c) Granulate. All rights reserved.
 # Licensed under the AGPL3 License. See LICENSE.md in the project root for license information.
 #
-from abc import ABC
-from typing import Iterable, Tuple, Type
+from granulate_utils.linux import get_kernel_release
+from granulate_utils.linux.kernel_messages import DefaultKernelMessagesProvider, EmptyKernelMessagesProvider
 
 from gprofiler.log import get_logger_adapter
-from gprofiler.utils import get_kernel_release
 
 logger = get_logger_adapter(__name__)
-kernel_release = get_kernel_release()
 
-KernelMessage = Tuple[float, int, str]
 
+class GProfilerKernelMessagesProvider(DefaultKernelMessagesProvider):
+    def on_missed(self):
+        logger.warning("Missed some kernel messages.")
 
-class Provider(ABC):
-    def iter_new_messages(self) -> Iterable[KernelMessage]:
-        ...
 
-
-class EmptyProvider(Provider):
-    def __init__(self):
-        print(
-            "This kernel does not support the new /dev/kmsg interface for reading messages,"
-            " or you lack the permissions for it."
-        )
+def get_kernel_messages_provider():
+    if get_kernel_release() < (3, 5):
```

Comment:
Shouldn't this logic move as well? Since it's going to be needed by whomever is going to use this API.

Reply:
Oh, I see that you have in https://github.com/Granulate/granulate-utils/pull/11. It is still cloned however.. Just for the logging?

Reply:
yes

Reply:
Why removed tho? I think the "No kernel messages monitoring" message is nice.

Reply:
readded